src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts

```diff
@@ -751,23 +755,23 @@ export class InstructorCourseEditPageComponent implements OnInit {
       name: formAddInstructor.controls.name.value,
       email: formAddInstructor.controls.email.value,
       role: formAddInstructor.controls.role.value,
-      isDisplayedToStudents: formAddInstructor.controls.isDisplayedToStudents.value,
+      isDisplayedToStudents: formAddInstructor.controls.isDisplayedToStudents.value.value,
       displayedName: formAddInstructor.controls.displayedName.value,
       privileges: this.getPrivilegesForRole(formAddInstructor.controls.role.value),
     };
 
     const paramsMap: { [key: string]: string } = {
       courseid: this.courseToEdit.id,
-      instructorname: addedInstructor.name,
-      instructoremail: addedInstructor.email,
-      instructorrole: addedInstructor.role,
-      instructordisplayname: addedInstructor.displayedName,
     };
 
-    const instructorIsDisplayed: string = 'instructorisdisplayed';
-    if (addedInstructor.isDisplayedToStudents) {
-      paramsMap[instructorIsDisplayed] = 'true';
-    }
+    const reqBody: InstructorCreateRequest = {
+      id: addedInstructor.googleId,
+      name: addedInstructor.name,
+      email: addedInstructor.email,
+      roleName: addedInstructor.role,
+      displayName: addedInstructor.displayedName,
+      isDisplayedToStudent: addedInstructor.isDisplayedToStudents != null,
```

Comment:
Why is the boolean check `!= null`?

Reply:
Yeah this looks odd. However, before this PR, the frontend actually passes a null value to the backend instead of false. The backend then checks if its null or not! 

I think the form makes it null if unticked?